### PR TITLE
WebXR Controls Improvements v2

### PR DESCRIPTION
**Pull request by eundersander**
## Motivation and Context

This is work from @BlaiseRitchie plus some cleanup. 6-DOF headset-tracking plus gamepad buttons for turning and moving forward. See original PR: https://github.com/facebookresearch/habitat-sim/pull/1076

## How Has This Been Tested

Tested with Ubuntu Chrome WebXR emulator, Android Chrome WebXR, and Quest 2 Oculus Browser WebXR.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/esp/bindings_js/bindings_js.cpp [hunk 1]
```diff
@@ -68,20 +91,10 @@ void Sensor_setLocalTransform(Sensor& sensor,
   node.setRotation(Magnum::Quaternion(quatf(rot)).normalized());
 }
 
-vec3f quaternionToEuler(const quatf& q) {
```
Comment (eundersander) [inline src/esp/bindings_js/bindings_js.cpp:71]: I don't see anything using these so I'm removing them.

File: src/esp/bindings_js/bindings_js.cpp [hunk 2]
```diff
@@ -131,6 +144,27 @@ EMSCRIPTEN_BINDINGS(habitat_sim_bindings_js) {
...
+      .class_function("yAxis", &Magnum::Vector3::yAxis)
+      .class_function("zAxis", &Magnum::Vector3::zAxis);
+
+  em::class_<Magnum::Quaternion>("Quaternion")
+      .constructor<Magnum::Vector3, float>()
+      .constructor<Magnum::Vector3>()
+      .function("normalized", &Magnum::Quaternion::normalized)
+      .function("inverted", &Magnum::Quaternion::inverted)
+      .function("transformVector", &Magnum::Quaternion::transformVector)
+      .class_function("mul", &Quaternion_mul)
```
Comment (eundersander) [inline src/esp/bindings_js/bindings_js.cpp:165]: `mul` class method instead of of `operator*`

File: src/esp/bindings_js/modules/web_demo.js [hunk 3]
```diff
@@ -31,10 +31,11 @@ class WebDemo {
 
     this.simenv.addAgent(agentConfig);
 
-    if (initializeTopDown) {
+    let topdownElement = document.getElementById("topdown");
+    if (initializeTopDown && topdownElement) {
```
Comment (eundersander) [inline src/esp/bindings_js/modules/web_demo.js:35]: make WebDemo robust to a missing topdown canvas

File: build_js.sh [hunk 4]
```diff
@@ -54,6 +54,6 @@ echo "python2 -m SimpleHTTPServer 8000"
 echo "Or:"
 echo "python3 -m http.server"
 echo "Then open in browser:"
-echo "http://0.0.0.0:8000/build_js/utils/viewer/viewer.html?scene=skokloster-castle.glb"
-echo "Or:"
 echo "http://0.0.0.0:8000/build_js/esp/bindings_js/bindings.html?scene=skokloster-castle.glb"
+echo "http://0.0.0.0:8000/build_js/esp/bindings_js/webvr.html?scene=skokloster-castle.glb"
```
Comment (Skylion007) [inline build_js.sh:58]: There should be a note here that this one requires a VR device.

**Review by Skylion007 — APPROVED** [review]

File: build_js.sh [hunk 5]
```diff
@@ -53,7 +53,7 @@ echo "Run:"
 echo "python2 -m SimpleHTTPServer 8000"
 echo "Or:"
 echo "python3 -m http.server"
-echo "Then open in browser:"
-echo "http://0.0.0.0:8000/build_js/utils/viewer/viewer.html?scene=skokloster-castle.glb"
```
Comment (eundersander) [inline build_js.sh:57]: I removed this reference to viewer.html. I consider it deprecated and it's not a good starting point for JS anymore.

**Pull request merged**